runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -59,6 +60,50 @@ class MemoryManager final {
     bool contains(void *ptr) { return _impl.contains(ptr); }
 };
 
+class SharedLibraryManager final {
+  private:
+    void *_handler{NULL};
+
+  public:
+    SharedLibraryManager() {}
+
+    SharedLibraryManager(std::string filename)
+    {
+        if (filename.find(".so") == std::string::npos) {
+            return;
```

Comment:
So we create the SharedLibManager successfully with invalid names, but raise an error if the name is valid but doesn't point to a shared object? Can we raise here an error as well to keep the semantics simpler?